src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -621,10 +636,22 @@ public UserDetails loadUserByUsername(String username)
             throw new UsernameNotFoundException("Using org*team format instead of username: " + username);
         }
 
+        User localUser = User.getById(username, false);
+
         Authentication token = SecurityContextHolder.getContext().getAuthentication();
 
         if (token == null) {
-            throw new UserMayOrMayNotExistException("Could not get auth token.");
+            if(localUser != null && GithubSecretStorage.contains(localUser)){
+                String accessToken = GithubSecretStorage.retrieve(localUser);
+                try {
+                    token = new GithubAuthenticationToken(accessToken, getGithubApiUri());
+                    SecurityContextHolder.getContext().setAuthentication(token);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
```

Comment:
Maybe `DataAccessException`?

Reply:
`DataAccessException` is for repository, this `IOException` is coming from a malformed URL, more a problem of configuration than runtime issue. Now I throw a `UserMayOrMayNotExistException` to be consistent with other plugin that cannot connect to the authentication server (like LDAP).